wasm/jit/jit_arm64_test.go

```diff
@@ -50,84 +49,84 @@ func (j *jitEnv) requireNewCompiler(t *testing.T) *arm64Compiler {
 	return ret
 }
 
-// TODO: delete this as this could be a duplication from other tests especially spectests.
-// Use this until we could run spectests on arm64.
-func TestArm64CompilerEndToEnd(t *testing.T) {
-	ctx := context.Background()
-	for _, tc := range []struct {
-		name string
-		body []byte
-		sig  *wasm.FunctionType
-	}{
-		{name: "empty", body: []byte{wasm.OpcodeEnd}, sig: &wasm.FunctionType{}},
-		{name: "br .return", body: []byte{wasm.OpcodeBr, 0, wasm.OpcodeEnd}, sig: &wasm.FunctionType{}},
-		{
-			name: "consts",
-			body: []byte{
-				wasm.OpcodeI32Const, 1, wasm.OpcodeI64Const, 1,
-				wasm.OpcodeF32Const, 1, 1, 1, 1, wasm.OpcodeF64Const, 1, 2, 3, 4, 5, 6, 7, 8,
-				wasm.OpcodeEnd,
-			},
-			// We push four constants.
-			sig: &wasm.FunctionType{Results: []wasm.ValueType{wasm.ValueTypeI32, wasm.ValueTypeI64, wasm.ValueTypeF32, wasm.ValueTypeF64}},
-		},
-		{
-			name: "add",
-			body: []byte{wasm.OpcodeI32Const, 1, wasm.OpcodeI32Const, 1, wasm.OpcodeI32Add, wasm.OpcodeEnd},
-			sig:  &wasm.FunctionType{Results: []wasm.ValueType{wasm.ValueTypeI32}},
-		},
-		{
-			name: "sub",
-			body: []byte{wasm.OpcodeI64Const, 1, wasm.OpcodeI64Const, 1, wasm.OpcodeI64Sub, wasm.OpcodeEnd},
-			sig:  &wasm.FunctionType{Results: []wasm.ValueType{wasm.ValueTypeI64}},
-		},
-		{
-			name: "mul",
-			body: []byte{wasm.OpcodeI64Const, 1, wasm.OpcodeI64Const, 1, wasm.OpcodeI64Mul, wasm.OpcodeEnd},
-			sig:  &wasm.FunctionType{Results: []wasm.ValueType{wasm.ValueTypeI64}},
-		},
-	} {
-		tc := tc
-		t.Run(tc.name, func(t *testing.T) {
-			engine := newEngine()
-			f := &wasm.FunctionInstance{
-				FunctionType: &wasm.TypeInstance{Type: tc.sig},
-				Body:         tc.body,
-			}
-			err := engine.Compile(f)
-			require.NoError(t, err)
-			_, err = engine.Call(ctx, f)
-			require.NoError(t, err)
-		})
-	}
-}
-
 func TestArchContextOffsetInEngine(t *testing.T) {
 	var eng engine
 	// If this fails, we have to fix jit_arm64.s as well.
 	require.Equal(t, int(unsafe.Offsetof(eng.jitCallReturnAddress)), engineArchContextJITCallReturnAddressOffset)
 }
 
 func TestArm64Compiler_returnFunction(t *testing.T) {
-	env := newJITEnvironment()
+	t.Run("exit", func(t *testing.T) {
+		env := newJITEnvironment()
 
-	// Build code.
-	compiler := env.requireNewCompiler(t)
-	err := compiler.emitPreamble()
-	require.NoError(t, err)
-	compiler.returnFunction()
+		// Build code.
```

Comment:
It is my guess eventually we'll have a test util that self-describes when when/where we do preamble, returnFunction, etc. so I'll hold off repetitive comment nits till then, as the test util would centralize such explanations